### ISSUE TITLE
Maintain order

### ISSUE THREAD

**Issue by bomeara** [ropensci/taxize#561]
Recently almost made a very bad mistake, assuming that the order of returned names with `best_match_only=TRUE` was the same as the input names. Not so. 

```
> gnr_resolve(names = c("Helianthus annuus", "Homo sapiens"), best_match_only=TRUE)
  user_supplied_name    submitted_name      matched_name data_source_title score
1  Helianthus annuus Helianthus annuus Helianthus annuus              NCBI 0.988
2       Homo sapiens      Homo sapiens      Homo sapiens              NCBI 0.988
```

which is in the example works fine, but

```
>  gnr_resolve(names = c("Homo sapiens", "Helianthus annuus"), best_match_only=TRUE)
  user_supplied_name    submitted_name      matched_name data_source_title score
1  Helianthus annuus Helianthus annuus Helianthus annuus              NCBI 0.988
2       Homo sapiens      Homo sapiens      Homo sapiens              NCBI 0.988
```

returns results in a new order.

I'd suggest maintaining the order or at least making a `warning()` that people shouldn't expect the same order to be returned.  I'm not sure if this is true for other resolvers, too.

**Comment by sckott** [comment]
thanks @bomeara  

I started to fix this in #559 in commit 3ae4ad4ecfb2058fe5428750ffc3b899f1090d48 but missed one route where it's still sorted. 

just pushed a change - reinstall `devtools::install_github("ropensci/taxize")` - should work now

**Comment by bomeara** [comment]
Yup, thanks, @sckott . The new code works great.
